web/packages/teleterm/src/deepLinks.ts

```diff
@@ -104,20 +117,41 @@ export function parseDeepLink(rawUrl: string): DeepLinkParseResult {
     };
   }
 
-  if (whatwgURL.pathname !== '/connect_my_computer') {
-    return { status: 'error', reason: 'unsupported-uri' };
-  }
-
-  const { host, hostname, port, username, pathname } = whatwgURL;
-  const url: DeepURL = {
+  // TODO (avatus): remove when authenticate web device case is implemented
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { host, hostname, port, username, pathname, searchParams } = whatwgURL;
+  const baseUrl: BaseDeepURL = {
     host,
     hostname,
     port,
     // whatwg-url percent-encodes usernames. We decode them here so that the rest of the app doesn't
     // have to do this. https://url.spec.whatwg.org/#set-the-username
     username: decodeURIComponent(username),
-    pathname,
   };
+  switch (pathname) {
+    case `/${Path.ConnectMyComputer}`: {
+      const url = { ...baseUrl, pathname };
+      return { status: 'success', url };
+    }
+    case `/${Path.AuthenticateWebDevice}`: {
+      // const id = searchParams.get('id');
+      // const token = searchParams.get('token');
+      // if (!(id && token)) {
+      //   return { status: 'error', reason: 'unsupported-uri' };
+      // }
 
-  return { status: 'success', url };
+      // const url = {
+      //   ...baseUrl,
+      //   pathname,
+      //   searchParams: {
+      //     id,
+      //     token,
+      //   },
+      // };
+      // return { status: 'success', url };
```

Comment:
Left this commented to get reviews on the overall change without actually implementing the case. The goal here is that we can construct a base URL and then use the switch case to build the rest of the "specific" URL (in this case, the CMC or device trust web) URL